sklearn/utils/testing.py

```diff
@@ -697,5 +705,27 @@ def check_skip_travis():
     if os.environ.get('TRAVIS') == "true":
         raise SkipTest("This test needs to be skipped on Travis")
 
+class TempMemMap(object):
+    def __init__(self, X):
+        self.temp_folder = tempfile.mkdtemp()
+        self.X = X
+        self.X_read_only = None
+
+    def __enter__(self):
+        fpath = op.join(self.temp_folder, 'data.pkl')
+        joblib.dump(self.X, fpath)
+        self.X_read_only = joblib.load(fpath, mmap_mode='r')
+        return self.X_read_only
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._clean()
+
+    def _clean(self):
+        del self.X_read_only
```

Comment:
Don't we if we want to be able to force garbage collection of X_read_only before removing temporary folder ?